plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -36,36 +33,52 @@ def service
       kubernetes_role.service_for(deploy_group) if has_service?
     end
 
-    def pretty_rc_doc(format: :json)
-      case format
-        when :json
-          JSON.pretty_generate(deployment_hash)
-        when :yaml, :yml
-          deployment_hash.to_yaml
-        else
-          deployment_hash.to_s
-      end
-    end
-
     def build
       kubernetes_release.try(:build)
     end
 
+    def release_doc_metadata
+      kubernetes_release.release_metadata.merge(role_metadata).merge(deploy_group_metadata)
+    end
+
+    def role_metadata
+      { role_id: kubernetes_role.id.to_s, role_name: kubernetes_role.name }
+    end
+
+    def deploy_group_metadata
+      { deploy_group_id: deploy_group.id.to_s, deploy_group_namespace: deploy_group.kubernetes_namespace }
+    end
+
     def nested_error_messages
       errors.full_messages
     end
 
-    def update_replica_count(new_count)
-      self.replicas_live = new_count
+    def update_release
+      kubernetes_release.update_status(self)
+    end
 
-      if replicas_live >= replica_target
-        self.status ='live'
-      elsif replicas_live > 0 && !failed?
-        self.status ='spinning_up'
+    def update_status(live_pods)
+      case
+      when live_pods == replica_target then self.status = :live
+      when live_pods.zero? then self.status = :dead
+      when live_pods > replicas_live then self.status = :spinning_up
+      when live_pods < replicas_live then self.status = :spinning_down
```

Comment:
Exactly, and that's the reason why I was wondering if we shouldn't just take this out, and update just the `deployed_at` timestamp indication when the deployment finishes. I've never seen a Pod crashing during a release, but assuming it can happen, how do we update the status? How can we distinguish a release that is spinning down from an ocasional crash? Maybe we can ignore it if it's the current release, but I wonder how many edge cases that we don't know yet will we get and if we really need this complexity.

We could simplify the states to just `deployed` and `failed_to_deploy`, and only update them at the end of the deployment. 

Food for thoughts when we come back to this?